packages/util-crypto/src/scrypt/fromU8a.ts

```diff
@@ -14,6 +14,15 @@ interface Result {
 }
 
 export function scryptFromU8a (data: Uint8Array): Result {
+  if (!(data instanceof Uint8Array)) {
+    throw new Error('Expected input to be a Uint8Array');
+  }
+
+  // Ensure the input is exactly 44 bytes: 32 for salt + 3 * 4 for N, p, r
+  if (data.length < 32 + 12) {
+    throw new Error(`Invalid input length: expected 44 bytes, found ${data.length}`);
+  }
+
   const salt = data.subarray(0, 32);
   const N = u8aToBn(data.subarray(32 + 0, 32 + 4), BN_LE_OPTS).toNumber();
```

Comment:
Can we get this additions replaced by the actual number? there is no point on doing those operations dinamically. If we really want to make it clear why we are using those numbers we can add a comment.
```suggestion
  const N = u8aToBn(data.subarray(32, 36), BN_LE_OPTS).toNumber();
```

Reply:
Done